contracts/rewards/SubgraphAvailabilityManager.sol

```diff
@@ -0,0 +1,206 @@
+     */
+    event VoteTimeLimitSet(uint256 voteTimeLimit);
+
+    /**
+     * @dev Emitted when an oracle votes to deny or allow a subgraph
+     */
+    event OracleVote(
+        bytes32 indexed subgraphDeploymentID,
+        bool deny,
+        uint256 oracleIndex,
+        uint256 timestamp
+    );
+
+    // -- Modifiers --
+
+    modifier onlyOracle(uint256 _oracleIndex) {
+        require(_oracleIndex < maxOracles, "SAM: index out of bounds");
+        require(msg.sender == oracles[_oracleIndex], "SAM: caller must be oracle");
+        _;
+    }
+
+    // -- Constructor --
+
+    /**
+     * @dev Contract constructor
+     * @param _governor Account that can set or remove oracles and set the vote time limit
+     * @param _rewardsManager Address of the RewardsManager contract
+     * @param _maxOracles Maximum number of oracles
+     * @param _executionThreshold Number of votes required to execute a deny or allow call to the RewardsManager
+     * @param _voteTimeLimit Vote time limit in seconds
+     */
+    constructor(
+        address _governor,
+        address _rewardsManager,
+        uint256 _maxOracles,
+        uint256 _executionThreshold,
+        uint256 _voteTimeLimit
+    ) {
+        require(_governor != address(0), "SAM: governor must be set");
+        require(_rewardsManager != address(0), "SAM: rewardsManager must be set");
+        require(_maxOracles > 1, "SAM: maxOracles must be greater than 1");
+        require(_executionThreshold > 1, "SAM: executionThreshold must be greater than 1");
+
+        Governed._initialize(_governor);
+        rewardsManager = IRewardsManager(_rewardsManager);
+
+        maxOracles = _maxOracles;
+        executionThreshold = _executionThreshold;
+        voteTimeLimit = _voteTimeLimit;
+        oracles = new address[](_maxOracles);
+
+        // Initialize oracles array with empty addresses
+        for (uint256 i = 0; i < _maxOracles; i++) {
+            oracles[i] = address(0);
+        }
+    }
+
+    // -- Functions --
+
+    /**
+     * @dev Set the vote time limit
+     * @param _voteTimeLimit Vote time limit in seconds
+     */
+    function setVoteTimeLimit(uint256 _voteTimeLimit) external onlyGovernor {
+        voteTimeLimit = _voteTimeLimit;
+        emit VoteTimeLimitSet(_voteTimeLimit);
+    }
+
+    /**
+     * @dev Set oracle address with index
+     * @param _index Index of the oracle
+     * @param _oracle Address of the oracle
+     */
+    function setOracle(uint256 _index, address _oracle) external onlyGovernor {
+        require(_index < maxOracles, "SAM: index out of bounds");
+        oracles[_index] = _oracle;
+        emit OracleSet(_index, _oracle);
+    }
+
+    /**
+     * @dev Vote deny or allow for a subgraph.
+     * NOTE: Can only be called by an oracle.
+     * @param _subgraphDeploymentID Subgraph deployment ID
+     * @param _deny True to deny, false to allow
+     * @param _oracleIndex Index of the oracle voting
+     */
+    function voteDenied(
```

Comment:
True, I was copying the interface for RewardsManager with `setDeny` but here makes more sense to just have `vote`. I'll update, thank you.

Reply:
Ah I forgot about that, I think we said we wanted to keep the interface the same in case the oracle doesn't need any code changes. 

What we can do is "porque no los dos?"

We can have `function vote()` as the main one with the logic and then `function voteDenied()` which is only an alias that then calls `vote()`

Reply:
Shouldn't it be `setDenied` to keep the same interface? (and infer the oracle index from msg.sender?)

Reply:
I was referring to keeping a naming convention. I believe it's better if we can update the interface and change it to `function vote(bytes32 _subgraphDeploymentID, bool _deny, uint256 _oracleIndex)` what do you think?